source/dotnet/wholesale-api/Edi.UnitTests/Validators/WholesaleServicesRequest/EnergySupplierValidatorTest.cs

```diff
@@ -0,0 +1,185 @@
+﻿// Copyright 2020 Energinet DataHub A/S
+//
+// Licensed under the Apache License, Version 2.0 (the "License2");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System.Diagnostics.CodeAnalysis;
+using Energinet.DataHub.Wholesale.Edi.Contracts;
+using Energinet.DataHub.Wholesale.Edi.UnitTests.Builders;
+using Energinet.DataHub.Wholesale.Edi.Validation;
+using Energinet.DataHub.Wholesale.Edi.Validation.WholesaleServicesRequest.Rules;
+using FluentAssertions;
+using FluentAssertions.Execution;
+using Xunit;
+
+namespace Energinet.DataHub.Wholesale.Edi.UnitTests.Validators.WholesaleServicesRequest;
+
+[SuppressMessage("Style", "VSTHRD200:Use \"Async\" suffix for async methods", Justification = "Async suffix is not needed for test methods")]
+public class EnergySupplierValidatorTest
+{
+    private const string ValidGlnNumber = "qwertyuiopasd"; // Must be 13 characters to be a valid GLN
+    private const string ValidEicNumber = "qwertyuiopasdfgh"; // Must be 16 characters to be a valid GLN
+
+    private static readonly ValidationError _invalidEnergySupplier = new("Feltet EnergySupplier skal være udfyldt med et valid GLN/EIC nummer når en elleverandør anmoder om data / EnergySupplier must be submitted with a valid GLN/EIC number when an energy supplier requests data", "E16");
+    private static readonly ValidationError _notEqualToRequestedBy = new("Elleverandør i besked stemmer ikke overenes med elleverandør i header / Energy supplier in message does not correspond with energy supplier in header", "E16");
+
+    private readonly EnergySupplierValidationRule _sut = new();
+
+    [Fact]
+    public async Task Validate_WhenEnergySupplierWithValidGlnNumber_ReturnsNoValidationErrors()
+    {
+        // Arrange
+        var message = new WholesaleServicesRequestBuilder()
+            .WithRequestedByActorId(ValidGlnNumber)
+            .WithRequestedByActorRole(DataHubNames.ActorRole.EnergySupplier)
+            .WithEnergySupplierId(ValidGlnNumber)
+            .Build();
+
+        // Act
+        var errors = await _sut.ValidateAsync(message);
+
+        // Assert
+        errors.Should().BeEmpty();
+    }
+
+    [Fact]
+    public async Task Validate_WhenEnergySupplierAndEnergySupplierIsValidEicNumber_ReturnsNoValidationErrors()
```

Comment:
```suggestion
    public async Task Validate_WhenEnergySupplierWithValidGlnNumber_ReturnsNoValidationErrors()
```
Since you renamed the test above, I would argue that this had the same "when" statement :)